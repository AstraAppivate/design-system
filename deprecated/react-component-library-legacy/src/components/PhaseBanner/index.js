import React from 'react'
import PropTypes from 'prop-types'

import Badge from '../Badge'

const PhaseBanner = ({ phase, children, link }) => (
  <div className={`rn-phase-banner rn-phase-banner--${phase}`}>
    <div className="rn-container">
      <Badge size="small" state="primary" type="solid">
        {phase}
      </Badge>
      <span className="rn-phase-banner__text">
        {children || (
          <>
            This is a new service, <a href={link}>Your feedback</a> will help to
            improve it
          </>
        )}
      </span>
    </div>
  </div>
)

PhaseBanner.propTypes = {
  /** The phase the banner is in */
  phase: PropTypes.node,
  /** The html to display with the banner */
  children: PropTypes.string,
  /** The link to use for the default html */
  link: PropTypes.string,
}

PhaseBanner.defaultProps = {
  phase: 'alpha',
  children: null,
  link: '/feedback',
}

export default PhaseBanner
