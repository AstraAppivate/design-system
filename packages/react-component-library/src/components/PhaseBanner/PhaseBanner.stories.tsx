import React from 'react'
import { Meta } from '@storybook/react/types-6-0'

import { PhaseBanner } from '.'

export default { component: PhaseBanner, title: 'Phase Banner' } as Meta

export const Default = (props: any) => <PhaseBanner {...props} />

export const Beta = () => <PhaseBanner phase="beta" />

Beta.storyName = 'Beta'

export const CustomLink = () => <PhaseBanner link="#" />

CustomLink.storyName = 'Custom link'

export const CustomText = () => (
  <PhaseBanner>
    Custom html can go here. <strong>This part is in bold!</strong>
  </PhaseBanner>
)

CustomText.storyName = 'Custom text'

export const FullWidth = () => <PhaseBanner isFullWidth />

FullWidth.storyName = 'Full width'
