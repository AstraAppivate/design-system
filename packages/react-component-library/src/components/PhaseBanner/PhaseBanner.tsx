import React from 'react'

import { BADGE_SIZE, BADGE_COLOR } from '../Badge'
import { StyledWrapper } from './partials/StyledWrapper'
import { StyledPhaseBanner } from './partials/StyledPhaseBanner'
import { StyledBadge } from './partials/StyledBadge'
import { StyledText } from './partials/StyledText'
import { ComponentWithClass } from '../../common/ComponentWithClass'

interface PhaseBannerProps extends ComponentWithClass {
  isFullWidth?: boolean
  link?: string
  phase?: 'alpha' | 'beta'
}

export const PhaseBanner: React.FC<PhaseBannerProps> = ({
  children,
  isFullWidth = false,
  link = '/feedback',
  phase = 'alpha',
  ...rest
}) => (
  <StyledPhaseBanner data-testid="phase-banner" {...rest}>
    <StyledWrapper
      $isFullWidth={isFullWidth}
      data-testid="phase-banner-wrapper"
    >
      <StyledBadge color={BADGE_COLOR.ACTION} size={BADGE_SIZE.SMALL}>
        {phase}
      </StyledBadge>
      <StyledText data-testid="phase-banner-content">
        {children || (
          <>
            This is a new service, <a href={link}>Your feedback</a> will help to
            improve it
          </>
        )}
      </StyledText>
    </StyledWrapper>
  </StyledPhaseBanner>
)

PhaseBanner.displayName = 'PhaseBanner'
