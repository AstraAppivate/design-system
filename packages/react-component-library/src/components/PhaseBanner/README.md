---
title: Phase Banner
description: A simple banner to indicate the phase of the project.
---

## Usage

The banner should be used directly underneath the main site header, before the content begins. If used with no props then it will look exactly like the example image below (linking to /feedback), however it can be customised to with any text or link that is required.

## Examples

<img src="images/phase-banner-example.gif" width="664" />

```
<Fragment>
  <PhaseBanner />
  <PhaseBanner link="/my-custom-page" />
  <PhaseBanner phase="beta" />
  <PhaseBanner>Custom html can go here.</PhaseBanner>
</Fragment>
```

<hr/>


## Properties

| Name      | Type           | Required | Default | Description                                                                                   
| --------- | -------------- | -------- | ------- | -----------
| phase     | String         | False    | `alpha` | (alpha/beta) Text to display in the phase banner badge
| link      | String         | False    | `/feedback`, | The link to use with the default message
| children  | Node           | False    | `This is a new service, <a href="">Your feedback</a> will help to improve it`, | HTML to display in the phase banner body

