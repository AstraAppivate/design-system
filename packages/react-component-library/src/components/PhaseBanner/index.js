import React from 'react'
import PropTypes from 'prop-types'

import Badge from '../Badge'

const PhaseBanner = ({ phase, html }) => (
  <PhaseBanner>
    <div className={`rn-phase-banner rn-phase-banner--phase-${phase}`}>
      <Badge size="small" state="primary" type="solid">
        {phase}
      </Badge>
      <span dangerouslySetInnerHTML={html} />
    </div>
  </PhaseBanner>
)

PhaseBanner.propTypes = {
  /** The phase the banner is in: Can be anything but specific
   * classes are applied to 'alpha' and 'beta'. */
  phase: PropTypes.string,
  /** The html to display with the banner */
  html: PropTypes.string,
}

PhaseBanner.defaultProps = {
  phase: 'alpha',
  html:
    'This is a new service, <a href="">Your feedback</a> will help to improve it',
}

export default PhaseBanner
