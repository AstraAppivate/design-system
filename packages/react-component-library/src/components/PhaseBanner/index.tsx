import React, { Fragment } from 'react'

import Badge from '../Badge'

interface PhaseBannerProps {
  phase?: 'alpha' | 'beta'
  link?: string
}

const PhaseBanner: React.FC<PhaseBannerProps> = ({
  phase = 'alpha',
  children,
  link = '/feedback',
}) => (
  <div className={`rn-phase-banner rn-phase-banner--${phase}`}>
    <div className="container">
      <Badge size="small" color="primary">
        {phase}
      </Badge>
      <span className="rn-phase-banner__text">
        {children || (
          <Fragment>
            This is a new service, <a href={link}>Your feedback</a> will help to
            improve it
          </Fragment>
        )}
      </span>
    </div>
  </div>
)

export default PhaseBanner
