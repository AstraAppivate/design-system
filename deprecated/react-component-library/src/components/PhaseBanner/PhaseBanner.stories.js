import React from 'react'
import { storiesOf } from '@storybook/react'

import PhaseBanner from './index'

const stories = storiesOf('PhaseBanner', module)

stories.add('Default', () => <PhaseBanner />)
stories.add('Custom phase tag', () => <PhaseBanner phase="beta" />)
stories.add('Custom link', () => <PhaseBanner link="/my-custom-link" />)
stories.add('Custom text', () => (
  <PhaseBanner>
    Custom html can go here. <strong>This part is in bold!</strong>
  </PhaseBanner>
))
