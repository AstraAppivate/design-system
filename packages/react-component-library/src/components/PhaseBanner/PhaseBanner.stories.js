import React from 'react'
import { storiesOf } from '@storybook/react'

import PhaseBanner from './index'

const stories = storiesOf('PhaseBanner', module)

stories.add('Default', () => <PhaseBanner />)
stories.add('Beta', () => <PhaseBanner phase="beta" />)
stories.add('Beta', () => <PhaseBanner phase="beta" />)
stories.add('Custom text', () => (
  <PhaseBanner html="Custom html can go here. <strong>This part is in bold!</strong>" />
))
