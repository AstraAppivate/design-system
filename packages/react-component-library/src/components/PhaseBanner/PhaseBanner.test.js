import React from 'react'
import { shallow } from 'enzyme'

import PhaseBanner from '.'

describe('PhaseBanner', () => {
  let wrapper

  describe('text', () => {
    beforeEach(() => {
      wrapper = shallow(<PhaseBanner />)
    })

    it('should render correctly render the AutoSuggest component', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
