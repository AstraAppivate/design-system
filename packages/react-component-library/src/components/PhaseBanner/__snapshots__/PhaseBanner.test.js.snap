// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhaseBanner text should render correctly render the AutoSuggest component 1`] = `
<div
  className="rn-phase-banner rn-phase-banner--alpha"
>
  <div
    className="container"
  >
    <Badge
      size="small"
      state="primary"
      type="solid"
      variation=""
    >
      alpha
    </Badge>
    <span
      className="rn-phase-banner__text"
      dangerouslySetInnerHTML={
        Object {
          "__html": "This is a new service, <a href=\\"\\">Your feedback</a> will help to improve it",
        }
      }
    />
  </div>
</div>
`;
